### PR TITLE
feat(aws-codepipeline): Pipeline now accepts existing IAM role

### LINKED ISSUE #2572: CodePipeline should allow you to pass an IAM Role
AWS CodePipeline doesn't currently support passing in an IAM Role when creating a pipeline, it [always creates a new one](https://github.com/awslabs/aws-cdk/blob/cfe46f697b1837bf61bec5b950494c26044282ff/packages/%40aws-cdk/aws-codepipeline/lib/pipeline.ts#L236-L238).

We have a generic CodePipeline IAM Role which we wish to share across multiple CodePipelines, currently we cannot do this.

### PR DESCRIPTION
**Pull request by AlexChesters**
Originally submitted as #2573, only this time without a horrible git history 😄 

Fixes #2572.

### Pull Request Checklist

* [N/A] Testing
  - As this feature is very similar to the existing pattern of [artifacts bucket](https://github.com/awslabs/aws-cdk/blob/cfe46f697b1837bf61bec5b950494c26044282ff/packages/%40aws-cdk/aws-codepipeline/lib/pipeline.ts#L224-L234), which isn't unit tested as far as I can see, I didn't add, modify or remove any existing unit tests.
* [X] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
  - __Design__: For significant features, design document added to `design` folder
* [X] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [N/A] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts [hunk 1]
```diff
@@ -68,6 +68,12 @@ export interface PipelineProps {
    */
   readonly artifactBucket?: s3.IBucket;
 
+  /**
+   * The IAM role to be assumed by this Pipeline.
+   * If not specified, a new IAM role will be created.
+   */
+  readonly role?: iam.Role;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts]: IRole

File: packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts [hunk 2]
```diff
@@ -68,6 +68,12 @@ export interface PipelineProps {
    */
   readonly artifactBucket?: s3.IBucket;
 
+  /**
+   * The IAM role to be assumed by this Pipeline.
+   * If not specified, a new IAM role will be created.
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts]: Use a `@default` marker in the docs to indicate the default behavior.

Comment (AlexChesters) [reply packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts]: @rix0rrr do you have an example of the `@default` format that's expected? I'm not familiar with it, and I can't see where this is used anywhere else in this file

**Comment by AlexChesters** [comment]
> Please add a test as well.
@rix0rrr sure 😄 - anything in particular that you'd want to be covered by this test? I'd copy from something like [`artifactBucket`](https://github.com/awslabs/aws-cdk/blob/553577a215a1a2ccc7ff2c16106a5c07f8600eee/packages/%40aws-cdk/aws-codepipeline/lib/pipeline.ts#L224-L234) but that doesn't appear to be tested currently

**Comment by rix0rrr** [comment]
> @rix0rrr sure smile - anything in particular that you'd want to be covered by this test

Yes, that if a role is passed in, it is referenced by the pipeline.

File: packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts (hunk 2 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts]: Here is an example:

https://github.com/awslabs/aws-cdk/blob/master/packages/@aws-cdk/aws-lambda/lib/function.ts#L115

I would not copy the second line "Both supplied and generated...". That feature is a pattern all over the library that can be assumed and doesn't have to be documented in every location.

**Comment by AlexChesters** [comment]
@rix0rrr cool, I'll get on that - thanks!

File: packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts (hunk 2 above)
Comment (AlexChesters) [reply packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts]: Gotcha, done that. How does it look?

**Comment by AlexChesters** [comment]
@rix0rrr I've added a test, mind having a look over and seeing what you think? I didn't think it fit into any of the existing test files so I created a new one

**Pull request merged**